src/api/AuthServer/AuthServer/Services/AuthService.cs

```diff
@@ -0,0 +1,68 @@
+﻿using AuthServer.DTO;
+using AutoMapper;
+using Microsoft.AspNetCore.Identity;
+
+namespace AuthServer.Services
+{
+    public class AuthService
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IConfiguration _configuration;
+        private readonly IMapper _mapper;
+
+        public AuthService(UserManager<ApplicationUser> userManager, IConfiguration configuration, IMapper mapper)
+        {
+            _userManager = userManager;
+            _configuration = configuration;
+            _mapper = mapper;
+        }
+
+        public async Task<UserDTO> Register(UserRegisterDTO userRegisterDTO)
+        {
+            var existingUser = await _userManager.FindByEmailAsync(userRegisterDTO.Email);
+            if (existingUser != null)
+            {
+                throw new InvalidOperationException("User with this email already exists");
+            }
+
+            if (userRegisterDTO.UserName == null)
+            {
+                userRegisterDTO.UserName = Guid.NewGuid().ToString();
+            }
```

Comment:
`Guid.NewGuid().ToString()` can be set as default value for `UserRegisterDto.UserName`. Or `userRegisterDTO.UserName ??= Guid.NewGuid().ToString()`